src/libs/DateUtils.js

```diff
@@ -324,6 +422,83 @@ function getStatusUntilDate(inputDate) {
     return translateLocal('statusPage.untilTime', {time: format(input, `${CONST.DATE.FNS_FORMAT_STRING} ${CONST.DATE.LOCAL_TIME_FORMAT}`)});
 }
 
+/**
+ * Update the hour and minute of a date.
+ *
+ * @param {string} inputTime - Time in "hh:mm A" format (like "10:55 AM").
+ * @param {string} inputDateTime - Date in "YYYY-MM-DD HH:mm:ss" format.
+ * @returns {string} - Date with updated time in "YYYY-MM-DD HH:mm:ss" format.
+ */
+const combineDateAndTime = (inputTime, inputDateTime) => {
+    // Parse the time from the input
+    const parsedTime = moment(inputTime, 'hh:mm A');
+
+    // Create a moment object from the input date-time (or just date)
+    const parsedDateTime = moment(inputDateTime, inputDateTime.includes(':') ? 'YYYY-MM-DD HH:mm:ss' : 'YYYY-MM-DD');
+
+    // Set the hour and minute from the time input to the date from the date-time input
+    parsedDateTime.set({
+        hour: parsedTime.get('hour'),
+        minute: parsedTime.get('minute'),
+    });
+
+    return parsedDateTime.format('YYYY-MM-DD HH:mm:ss');
+};
+
+/**
+ * @param {String} time
```

Comment:
The purpose of the function is a bit unclear just looking at it, so could do with a brief description or the function name updated. (I'm unsure if the point is to output an object or to convert some input time to 12hr).

Reply:
We use it for parsing date like `'YYYY-MM-DD HH:mm:ss'` to the initial value for TimePicker, which consume each value separately (hour, minute, am/pm)

Reply:
Ah, gotcha! Maybe it could be renamed `getTimeObjectFromDate` so that it's a bit closer to what it actually does? Not sure if we need to specify 12 hour in the name if there isn't a corresponding 24-hr one.

Reply:
This 12-hour format is confusing. I would leave a reference in the name if we switch it to 24-hour and refactor these functions.
What about `getTime12HourObjectFromDate`?